File: WePeiYang/lostfound/src/main/java/com/twtstudio/tjwhm/lostfound/release/ReleaseTableAdapter.java
package com.twtstudio.tjwhm.lostfound.release;

import android.content.Context;
import android.graphics.Color;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.twtstudio.tjwhm.lostfound.R;
import com.twtstudio.tjwhm.lostfound.R2;
import com.twtstudio.tjwhm.lostfound.support.Utils;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by tjwhm on 2017/7/8.
 **/

public class ReleaseTableAdapter extends RecyclerView.Adapter {
    Context context;
    int positionSelected;
    ReleaseContract.ReleaseView releaseView;

    public ReleaseTableAdapter(Context context, int positionSelected, ReleaseContract.ReleaseView releaseView) {
        this.context = context;
        this.positionSelected = positionSelected;
        this.releaseView = releaseView;
    }

    public class ReleaseViewHolder extends RecyclerView.ViewHolder {
        @BindView(R2.id.release_item_type)
        TextView release_item_type;
        @BindView(R2.id.release_item_cardview)
        CardView release_item_cardview;

        public ReleaseViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view;
        view = LayoutInflater.from(context).inflate(R.layout.lf_item_release_type, parent, false);

        return new ReleaseViewHolder(view);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        ReleaseViewHolder viewHolder = (ReleaseViewHolder) holder;
        viewHolder.release_item_type.setText(Utils.getType(position + 1));
        viewHolder.release_item_cardview.setCardBackgroundColor(Color.parseColor("#dfdfdf"));
        if (position == positionSelected) {
            viewHolder.release_item_cardview.setCardBackgroundColor(Color.parseColor("#00a1e9"));
        }
        viewHolder.itemView.setOnClickListener(view -> {
            releaseView.drawRecyclerView(position);
            releaseView.onTypeItemSelected(position);
        });

    }

    @Override
    public int getItemCount() {
        return 13;
    }
}

File: WePeiYang/lostfound2/src/main/java/com/yookiely/lostfond2/release/ReleaseTableAdapter.kt
package com.yookiely.lostfond2.release

import android.content.Context
import android.graphics.Color
import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.example.lostfond2.R
import com.yookiely.lostfond2.service.Utils
import org.jetbrains.anko.textColor

class ReleaseTableAdapter(val context: Context, val positionSelected: Int,
                          val releaseView: ReleaseContract.ReleaseView) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    class ReleaseViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val releaseItemType = itemView.findViewById<TextView>(R.id.release_item_type)!!
        val releaseItemCardView = itemView.findViewById<CardView>(R.id.release_item_cardview)!!
        val releaseItemSuperscript = itemView.findViewById<ImageView>(R.id.release_item_superscript)!!
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.lf2_item_release_type, parent, false)

        return ReleaseViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val viewHolder = holder as ReleaseViewHolder
        viewHolder.apply {
            releaseItemType.text = Utils.getType(position + 1)

            if (position == positionSelected) {
                releaseItemType.textColor = Color.parseColor("#00A1EA")
                releaseItemCardView.setCardBackgroundColor(Color.parseColor("#BEDFEE"))
                releaseItemSuperscript.visibility = View.VISIBLE
            } else {
                releaseItemType.textColor = Color.parseColor("#333333")
                releaseItemCardView.setCardBackgroundColor(Color.parseColor("#DFDFDF"))
                releaseItemSuperscript.visibility = View.GONE
            }
        }

        viewHolder.itemView.setOnClickListener {
            run {
                releaseView.apply {
                    drawRecyclerView(position)
                    onTypeItemSelected(position)
                }
            }
        }

    }

    override fun getItemCount() = 13
}